x-pack/plugins/apm/public/hooks/useFetcher.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { flatten, isObject } from 'lodash';
+import { useEffect, useState } from 'react';
+
+export enum FETCH_STATUS {
+  LOADING = 'loading',
+  SUCCESS = 'success',
+  FAILURE = 'failure'
+}
+
+export function useFetcher<Args extends any[], Response>(
+  fn: (...args: Args) => Promise<Response>,
+  fnArgs: Args
+) {
+  const [status, setStatus] = useState<FETCH_STATUS | null>(null);
+  const [data, setData] = useState<Response | null>(null);
```

Comment:
I thought about it, but felt it was unnecessary since we can do this:

```js
const { data = initialState } = useFetcher(loadUsers, []);
```

which is more intuitive (to me) than something like:
```js
const { data } = useFetcher(loadUsers, [], initialState);
```

Reply:
Oh I do like that better but I'm not sure that works unless we default to `undefined` inside useFetcher ... i.e.:
```
function abc(x = 'fallback') {
  console.log(x)
}

let y = undefined;

abc() // "fallback"
abc(y) // "fallback"
abc(null) // null
```

Reply:
Oh, good point. I remember I changed it from `null` to `undefined` and thought it was in this PR. Turns out it is in the next: https://github.com/sqren/kibana/pull/11/files#diff-992fb053a520373d6e1eecaab5d31653R22